packages/runtime/container-runtime/src/gc/gcTelemetry.ts

```diff
@@ -119,31 +124,70 @@ export class GCTelemetryTracker {
 			return false;
 		}
 
+		// Non-tombstone events are logged once per event per node. A unique id is generated by joining
+		// node state (inactive / sweep ready), node's id and usage (loaded / changed / revived).
 		if (this.loggedUnreferencedEvents.has(uniqueEventId)) {
 			return false;
 		}
 		return true;
 	}
 
 	/**
-	 * Called when a node is used. If the node is not active, log an event indicating object is used when its not active.
+	 * Called when a node is used. If the node is not active or tombstoned, log telemetry indicating object is used
+	 * when it should not have been.
 	 */
 	public nodeUsed(nodeUsageProps: INodeUsageProps) {
 		// If there is no reference timestamp to work with, no ops have been processed after creation. If so, skip
 		// logging as nothing interesting would have happened worth logging.
-		// If the node is not unreferenced, skip logging.
-		const nodeStateTracker = this.getNodeStateTracker(nodeUsageProps.id);
-		if (!nodeStateTracker || nodeUsageProps.currentReferenceTimestampMs === undefined) {
+		if (nodeUsageProps.currentReferenceTimestampMs === undefined) {
 			return;
 		}
 
-		// We log these events once per event per node. A unique id is generated by joining node state (inactive / sweep ready),
-		// node's id and usage (loaded / changed / revived).
-		const uniqueEventId = `${nodeStateTracker.state}-${nodeUsageProps.id}-${nodeUsageProps.usageType}`;
+		const nodeStateTracker = this.getNodeStateTracker(nodeUsageProps.id);
 		const nodeType = this.getNodeType(nodeUsageProps.id);
+		const {
+			usageType,
+			currentReferenceTimestampMs,
+			packagePath,
+			id: untaggedId,
+			fromId: untaggedFromId,
+			...propsToLog
+		} = nodeUsageProps;
+		const unrefEventProps: Omit<IUnreferencedEventProps, "state" | "usageType"> = {
+			type: nodeType,
+			unrefTime: nodeStateTracker?.unreferencedTimestampMs ?? -1,
+			age:
+				nodeStateTracker !== undefined
+					? nodeUsageProps.currentReferenceTimestampMs -
+					  nodeStateTracker.unreferencedTimestampMs
+					: -1,
+			timeout:
+				nodeStateTracker?.state === UnreferencedState.Inactive
+					? this.configs.inactiveTimeoutMs
+					: this.configs.sweepTimeoutMs,
+			...tagCodeArtifacts({ id: untaggedId, fromId: untaggedFromId }),
+			...propsToLog,
+			...this.createContainerMetadata,
+		};
+
+		// If the node that is used is tombstoned, log a tombstone telemetry.
+		// Note that this is done before checking if "nodeStateTracker" is undefined below because unreferenced
+		// tracking may not have yet been enabled. That happens only after the client transitions to write mode.
+		if (nodeUsageProps.isTombstoned) {
+			this.logTombstoneUsageTelemetry(nodeUsageProps, unrefEventProps, nodeType, usageType);
```

Comment:
Early return here...?

Reply:
I don't think we should do that because it will skip logging inactive telemetry which might be important.